src/pages/act-now.js

```diff
@@ -22,7 +19,16 @@ function ActNow(props) {
   return <ActNowLoggedInPage {...props} />;
 }
 
-export async function getStaticProps({ locale }) {
+ActNow.propTypes = {
+  session: PropTypes.shape({}),
+};
+
+ActNow.defaultProps = {
+  session: null,
+};
```

Comment:
Why are we passing `session` like this?

Reply:
to prevent the page flashing whiles checking the session, so it's computed in the `getServerSideProps` @kilemensi

Reply:
Nah, you **should** return `session` from `getServerSideProps` but that one goes to the [`SessionProvider`](https://github.com/CodeForAfrica/PromiseTracker/blob/fe66f36cb716d531b9247bfb0a5c4d4dea9938dd/src/pages/_app.js#L29) which prevents the page flashing.

On the client you just use `useSession` as you'd normally do.